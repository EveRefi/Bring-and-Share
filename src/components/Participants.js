import React, { useState, useEffect } from "react";
import axios from "axios";
import ParticipantCard from "../components/ParticipantCard";
import "../styles/Participants.css";

const Participants = ({ eventId, fetchParticipants, setFetchParticipants }) => {
  const [participants, setParticipants] = useState([]);
  useEffect(() => {
    async function fetchData() {
      await axios
        .get(
          `https://final-mcrcodes-project.herokuapp.com/events/${eventId}/participants`
        )
        .then((response) => setParticipants(response.data))
        .catch((err) => console.log(err));
    }

    fetchData();
  }, [eventId, fetchParticipants]);

  const removeParticipant = (id) => {
    axios
      .delete(
        `https://final-mcrcodes-project.herokuapp.com/events/${eventId}/participants/${id}`
      )
      .then(() => {
        setFetchParticipants(!fetchParticipants);
      })
      .catch((err) => console.log(err));
  };
  return (
    <div data-testid="participants" className="participants">
      {participants.map((person, index) => (
        <ParticipantCard
          key={index}
          name={person.name}
          email={person.email}
          toBring={person.toBring}
          dislikes={person.dislikes}
          dietInfo={person.dietInfo}
          id={person.id}
          removeParticipant={() => removeParticipant(person.id)}
        />
      ))}
    </div>
  );
};

export default Participants;
