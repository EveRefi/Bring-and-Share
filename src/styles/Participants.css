.participants {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  border: 5px solid lightcoral;
}
