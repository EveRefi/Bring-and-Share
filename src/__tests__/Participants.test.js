import React from "react";
import { render } from "@testing-library/react";
import Participants from "../components/Participants";

describe('Partcipants components', () =>{
    const mockId = 123

    it('matches the snapshot', () => {
        const {asFragment} = render (<Participants eventId={mockId}/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('returns the right elements', () => {
        const {getByTestId} = render (<Participants eventId={mockId}/>)

        expect(getByTestId('participants')).toBeInTheDocument()
    })
})