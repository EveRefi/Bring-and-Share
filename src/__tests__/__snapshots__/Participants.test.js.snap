// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partcipants components matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="participants"
    data-testid="participants"
  />
</DocumentFragment>
`;
